titus-server-runtime/src/main/java/com/netflix/titus/runtime/jobmanager/gateway/SanitizingJobServiceGateway.java

```diff
@@ -73,4 +79,30 @@ public Completable updateJobCapacity(JobCapacityUpdate jobCapacityUpdate) {
         }
         return delegate.updateJobCapacity(jobCapacityUpdate);
     }
+
+    @Override
+    public Completable updateJobCapacityWithOptionalAttributes(JobCapacityUpdateWithOptionalAttributes jobCapacityUpdateWithOptionalAttributes) {
+        final JobCapacityWithOptionalAttributes jobCapacityWithOptionalAttributes = jobCapacityUpdateWithOptionalAttributes.getJobCapacityWithOptionalAttributes();
+        return delegate.findJob(jobCapacityUpdateWithOptionalAttributes.getJobId())
+                .map(j -> {
+                    com.netflix.titus.api.jobmanager.model.job.JobDescriptor jobDescriptor = V3GrpcModelConverters.toCoreJobDescriptor(j.getJobDescriptor());
+                    if (!JobFunctions.isServiceJob(jobDescriptor)) {
+                        throw JobManagerException.notServiceJob(j.getId());
+                    }
+                    ServiceJobExt serviceJobExt = (ServiceJobExt) jobDescriptor.getExtensions();
+                    Capacity.Builder newCapacityBuilder = serviceJobExt.getCapacity().toBuilder();
+                    Evaluators.acceptIfTrue(jobCapacityWithOptionalAttributes.hasDesired(), valueAccepted -> newCapacityBuilder.withDesired(jobCapacityWithOptionalAttributes.getDesired().getValue()));
+                    Evaluators.acceptIfTrue(jobCapacityWithOptionalAttributes.hasMax(), valueAccepted -> newCapacityBuilder.withMax(jobCapacityWithOptionalAttributes.getMax().getValue()));
+                    Evaluators.acceptIfTrue(jobCapacityWithOptionalAttributes.hasMin(), valueAccepted -> newCapacityBuilder.withMin(jobCapacityWithOptionalAttributes.getMin().getValue()));
+
+                    return newCapacityBuilder.build();
+                })
+                .flatMapCompletable(newCapacity -> {
+                    Set<ValidationError> violations = entitySanitizer.validate(newCapacity);
```

Comment:
If you change the core models to be always optional, this may be simplified to validate only what's needed without requiring a to fetch current values first